src/contracts/core/AllocationManager.sol

```diff
@@ -55,138 +57,139 @@ contract AllocationManager is
-            // proprtional magnitudes relative to each other.
-            uint64 maxMagnitude = _maxMagnitudeHistory[msg.sender][allocation.strategy].latest();
-            require(maxMagnitude == allocation.expectedMaxMagnitude, InvalidExpectedMaxMagnitude());
-
-            // 2. For the given (operator,strategy) clear any clearable pending deallocations to free up encumberedMagnitude
-            _clearDeallocationQueue({operator: msg.sender, strategy: allocation.strategy, numToClear: type(uint16).max});
-
-            for (uint256 j = 0; j < allocation.operatorSets.length; ++j) {
-                bytes32 operatorSetKey = _encodeOperatorSet(allocation.operatorSets[j]);
-
-                // Ensure there is not already a pending modification
-                PendingMagnitudeInfo memory info =
-                    _getPendingMagnitudeInfo(msg.sender, allocation.strategy, operatorSetKey);
-                require(info.pendingDiff == 0, ModificationAlreadyPending());
-
-                info.pendingDiff = _calcDelta(info.currentMagnitude, allocation.magnitudes[j]);
-                require(info.pendingDiff != 0, SameMagnitude());
-
-                // Calculate the effectTimestamp for the modification
-                if (info.pendingDiff < 0) {
-                    info.effectTimestamp = uint32(block.timestamp) + DEALLOCATION_DELAY;
-
-                    // Add the operatorSet to the deallocation queue
-                    deallocationQueue[msg.sender][allocation.strategy].pushBack(operatorSetKey);
-                } else if (info.pendingDiff > 0) {
-                    info.effectTimestamp = uint32(block.timestamp) + operatorAllocationDelay;
-
-                    // For allocations, immediately add to encumberedMagnitude to ensure the operator
-                    // can't allocate more than their maximum
-                    info.encumberedMagnitude = _addInt128(info.encumberedMagnitude, info.pendingDiff);
-                    require(info.encumberedMagnitude <= maxMagnitude, InsufficientAllocatableMagnitude());
+        for (uint256 i = 0; i < params.length; i++) {
+            require(params[i].strategies.length == params[i].newMagnitudes.length, InputArrayLengthMismatch());
+
+            // Check that the operator set exists and get the operator's registration status
+            // Operators do not need to be registered for an operator set in order to allocate
+            // slashable magnitude to the set. In fact, it is expected that operators will
+            // allocate magnitude before registering, as AVS's will likely only accept
+            // registrations from operators that are already slashable.
+            OperatorSet calldata operatorSet = params[i].operatorSet;
+            bool isRegistered = _isRegistered(msg.sender, operatorSet);
+            require(_operatorSets[operatorSet.avs].contains(operatorSet.id), InvalidOperatorSet());
```

Comment:
Similar to other comment, the `_isRegistered(msg.sender, operatorSet)` call being true already implies existence of the operatorSet. We could remove this require check

Reply:
I like this on as-is because `isRegistered` isn't a requirement to allocate magnitude. However, the operator set existing should be a requirement!